### PR TITLE
Added a 3.2 home task StorageOfContacts + added a 3.3 homework: work with delegates

### PR DESCRIPTION
**Pull request by Vitaliiitk**

### CODE REVIEW

File: 3.2ListOfContacts/ListOfContacts/ContactStorage.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                {
+                    phoneBook[keys[0]].Add(contact);
+                }
+
+                else if (IsUkrainian(nameAndSurname) == true && Regex.IsMatch(phoneNumber, @"^\u002b\d{12}") && phoneNumber.StartsWith("+38"))
+                {
+                    phoneBook[keys[1]].Add(contact);
+                }
+
+                else if (IsEnglish(nameAndSurname) == false && IsUkrainian(nameAndSurname) == false && Regex.IsMatch(phoneNumber, @"^\u002b\d{12}") && phoneNumber.StartsWith("+"))
```
Comment (vladboss61) [inline 3.2ListOfContacts/ListOfContacts/ContactStorage.cs]: IsEnglish(nameAndSurname) == false replace with !IsEnglish(nameAndSurname)

File: 3.2ListOfContacts/ListOfContacts/ContactStorage.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            };
+
+            try
+            {
+                if (IsEnglish(nameAndSurname) == true && Regex.IsMatch(phoneNumber, @"^\u002b\d{12}") && phoneNumber.StartsWith("+44"))
+                {
+                    phoneBook[keys[0]].Add(contact);
+                }
+
+                else if (IsUkrainian(nameAndSurname) == true && Regex.IsMatch(phoneNumber, @"^\u002b\d{12}") && phoneNumber.StartsWith("+38"))
```
Comment (vladboss61) [inline 3.2ListOfContacts/ListOfContacts/ContactStorage.cs]: IsUkrainian(nameAndSurname) == true -> IsUkrainian(nameAndSurname)

File: 3.2ListOfContacts/ListOfContacts/ContactStorage.cs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        {
+            Contact contact = new Contact()
+            {
+                NameAndSurname = nameAndSurname,
+                PhoneNumber = phoneNumber
+            };
+
+            try
+            {
+                if (IsEnglish(nameAndSurname) == true && Regex.IsMatch(phoneNumber, @"^\u002b\d{12}") && phoneNumber.StartsWith("+44"))
```
Comment (vladboss61) [inline 3.2ListOfContacts/ListOfContacts/ContactStorage.cs]: IsUkrainian(nameAndSurname) == true -> IsUkrainian(nameAndSurname)

File: 3.2ListOfContacts/ListOfContacts/Contact.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ListOfContacts
+{
+    internal class Contact
```
Comment (vladboss61) [inline 3.2ListOfContacts/ListOfContacts/Contact.cs]: add `sealed`

File: 3.2ListOfContacts/ListOfContacts/App.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ListOfContacts
+{
+    internal class App
+    {
+        public void Run()
+        {
+            ContactStorage cont = new ContactStorage();
```
Comment (vladboss61) [inline 3.2ListOfContacts/ListOfContacts/App.cs]: cont -> contacts

File: 3.2ListOfContacts/ListOfContacts/ContactStorage.cs [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace ListOfContacts
+{
+    public sealed class ContactStorage
+    {
+
+        private readonly List<string> keys = new List<string>()
```
Comment (vladboss61) [inline 3.2ListOfContacts/ListOfContacts/ContactStorage.cs]: It is better just use constants like 
public const English = 'English' and so on,

File: 3.2ListOfContacts/ListOfContacts/ContactStorage.cs (hunk 1 above)
Comment (Vitaliiitk) [reply 3.2ListOfContacts/ListOfContacts/ContactStorage.cs]: I think, I fixed this.

File: 3.2ListOfContacts/ListOfContacts/ContactStorage.cs (hunk 2 above)
Comment (Vitaliiitk) [reply 3.2ListOfContacts/ListOfContacts/ContactStorage.cs]: I think, I fixed this.

File: 3.2ListOfContacts/ListOfContacts/ContactStorage.cs (hunk 3 above)
Comment (Vitaliiitk) [reply 3.2ListOfContacts/ListOfContacts/ContactStorage.cs]: I think, I fixed this.

File: 3.2ListOfContacts/ListOfContacts/Contact.cs (hunk 4 above)
Comment (Vitaliiitk) [reply 3.2ListOfContacts/ListOfContacts/Contact.cs]: I think, I fixed this.

File: 3.2ListOfContacts/ListOfContacts/App.cs (hunk 5 above)
Comment (Vitaliiitk) [reply 3.2ListOfContacts/ListOfContacts/App.cs]: I think, I fixed this.

File: 3.2ListOfContacts/ListOfContacts/ContactStorage.cs (hunk 6 above)
Comment (Vitaliiitk) [reply 3.2ListOfContacts/ListOfContacts/ContactStorage.cs]: I think, I fixed this.

**Pull request merged**